packages/context/src/app/AppProvider.tsx

```diff
@@ -62,8 +78,57 @@ export type AppProviderProps = {
 
 function AppProvider(props: AppProviderProps) {
     const { app, children } = props;
-
-    return <AppContext.Provider value={app}>{children}</AppContext.Provider>;
+    const apiServices = useApiServices();
+
+    const appWithFunctions: AppWithFunctions = {
+        ...app,
+        /**
+         * Looks up the default page slug for a given object or its nearest type ancestor.
```

Comment:
Comment should be part of the type definition, i.e. it should show up in the generated `*.d.ts` file.